test/util/discovery/discovery.go

```diff
@@ -0,0 +1,72 @@
+package discovery
+
+import (
+	"context"
+	"os"
+	"strconv"
+
+	sriovv1 "github.com/openshift/sriov-network-operator/pkg/apis/sriovnetwork/v1"
+	"github.com/openshift/sriov-network-operator/test/util/client"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	runtimeclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// Enabled indicates whether test discovery mode is enabled.
+func Enabled() bool {
+	discoveryMode, _ := strconv.ParseBool(os.Getenv("DISCOVERY_MODE"))
+	return discoveryMode
+}
+
+func getSriovNodes(clients *client.ClientSet, sriovNodes []string) ([]corev1.Node, error) {
+	allNodes, err := clients.Nodes().List(context.Background(), metav1.ListOptions{
+		LabelSelector: "node-role.kubernetes.io/worker",
+	})
+	if err != nil {
+		return nil, err
+	}
+	var nodes []corev1.Node
+	for _, node := range allNodes.Items {
+		if containsNode(node.ObjectMeta.Name, sriovNodes) {
+			nodes = append(nodes, node)
+		}
+	}
+	return nodes, nil
+}
+
+func containsNode(name string, sriovNodes []string) bool {
+	for _, node := range sriovNodes {
+		if node == name {
+			return true
+		}
+	}
+	return false
+}
+
+// DiscoveredResources discovers resources needed by the tests in discovery mode
+func DiscoveredResources(clients *client.ClientSet, sriovNodes []string, operatorNamespace string) (preferedNode string, preferedResourceName string, preferedResourceCount int, err error) {
+	policyList := sriovv1.SriovNetworkNodePolicyList{}
+	err = clients.List(context.Background(), &policyList, runtimeclient.InNamespace(operatorNamespace))
+	if err != nil {
+		return "", "", 0, err
+	}
+	nodes, err := getSriovNodes(clients, sriovNodes)
+	if err != nil {
+		return "", "", 0, err
```

Comment:
Either use named returns or not. I prefer not using them, but it's just a matter of taste.
Here you can just say "return" because err is set and the others are 0 values.

Reply:
Done